src/main/java/org/openpnp/machine/reference/vision/ReferenceBottomVision.java

```diff
@@ -72,10 +74,43 @@ public PartAlignmentOffset findOffsets(Part part, BoardLocation boardLocation,
 
         // Pre-rotate to minimize runout
         double preRotateAngle = 0;
-        if (preRotate && boardLocation != null && placementLocation != null) {
-            preRotateAngle =
-                    Utils2D.calculateBoardPlacementLocation(boardLocation, placementLocation)
-                           .getRotation();
+        if (preRotate) {
+			if(part == null || nozzle.getPart() == null) { throw new Exception("no part on nozzle"); }			
+			if((!part.toString().equals(nozzle.getPart().toString()))) { throw new Exception("Part mismatch with part on nozzle"); }
```

Comment:
Note: It's better to compare part.getId() or even just part == part (since there can be only one instance of a part at a time). Comparing toString() is not recommended since toString() is intended to be human readable and informative, and may change over time.